### DISCUSSION TITLE
Artisan roasting software

### DISCUSSION THREAD

**Discussion by rajivpal96** [artisan-roaster-scope/artisan#673] [Q&A]
Has anyone had experience interfacing artisan software with a Joper Roaster 30Kg roaster using Yokogawa UP550 controller with Modbus interface?

**Comment by MAKOMO** [comment]
Assuming you have the correct serial driver installed on your computer that can connect to the RS485 to serial interface and the UP550 is configured to use its default serial parameters (you might want to check the devices configuration on this) as I found in the modules documentation:

https://web-material3.yokogawa.com/IM05E01C02-46E_003.pdf

and its communication specification:

https://web-material3.yokogawa.com/IM05G01B02-02E_007_1.pdf

baud: 9600
parity: Even
stopbits: 1
databits: 8
slaveID: 1

you find the two process values in the MODBUS addresses

PV.1 : 40003
PV.2 : 40019

those translate to holding registers 3 and 19 (or 2 and 18, or 4 and 20) which can be read via MODBUS function 3. So your configuration might need to look like the following (modulo the selection of the correct serial port as assigned by your serial driver):

<img width="1083" alt="Screenshot 2021-08-26 at 14 12 02" src="https://user-images.githubusercontent.com/13909011/130961280-d92d66b3-1a40-4e61-b8cf-d97e71e19f42.png">

<img width="1072" alt="Screenshot 2021-08-26 at 14 12 09" src="https://user-images.githubusercontent.com/13909011/130961296-da6f59d8-0d4c-483a-a8c1-a060f5f62c24.png">
